app/api/companies.py

```diff
@@ -138,24 +140,41 @@ def get_company(id):
 def add_company():
     data = request.get_json() or {}
 
-    # Validation
-    if 'company_id' in data:
-        return bad_request("Create company cannot include company_id. For modifying existing companies please use the PUT method")
-    if 'company_name' not in data:
-        return bad_request("Must include company_name field")
-    if Companies.query.filter_by(company_name=data['company_name']).first():
-        return bad_request("A company with that name already exists, please use another name")
-
-    company = Companies()
-    # TODO: Finalize the from_dict method
-    company.from_dict_new(data)
-    db.session.add(company)
-    db.session.commit()
-
-    response = jsonify(company.to_dict())
+    try:
+        result = CompaniesValidationSchema().load(data)
+    except ValidationError as err:
+        print(err.messages)
+        print(err.valid_data)
+        return bad_request(err.messages)
+
+    # TODO: Validate if company name is already in DB?
+
+    new_company = Companies(company_name=result['company_name'], logo_image_src=result['logo_image_src'],
+                            website=result['website'], year=result['year'], company_size=result['company_size'])
+
+    # Check if city is already in DB:
```

Comment:
So this is a pretty "standard" problem you might have with related record: do they exist -> add them, don't they exist -> create them. 

Some ORM's have a method for this like `findOrCreate`, but SQL doesn't seem to have that: [https://stackoverflow.com/questions/2546207/does-sqlalchemy-have-an-equivalent-of-djangos-get-or-create](according to stackoverflow)

But you can make your own method that does this. So you could do something like: 

```python
city = Cities.get_or_create(session, city_name=result['city_name'])
new_company.city_id = city.city_id
db.session.add(new_company)
db.session.commit()
```

or something along those lines

Reply:
Good idea! Will be added in the next commit to this PR.